src/Microsoft.OData.Core/ODataMessageReaderSettings.cs

```diff
@@ -46,6 +46,7 @@ public ODataMessageReaderSettings(ODataVersion odataVersion)
             this.EnableCharactersCheck = false;
             this.Version = odataVersion;
             this.LibraryCompatibility = ODataLibraryCompatibility.Latest;
+            this.EnablePropertyNameCaseInsensitive = false; // for back-compatible

```

Comment:
@xuzhg Will we make case-insensitive comparison the default in 8.x major release? If yes, maybe you could create an issue to track that and tag it appropriately